File: eww/bat.py
#!/usr/bin/pypy3

import subprocess
import json 

def battperc():
    battperc = subprocess.run(
        ["acpi", "-b"],
        capture_output=True,
        text=True,
    ).stdout

    [perc, timerem] = battperc.split(',')[1:3]
    perc = perc.strip().replace('%', '')
    timerem = timerem.strip().replace('remaining', '').replace('until charged', '').strip().replace(' ', '')
    return (int(perc), timerem[:-3])

def batclass(perc):
    if 0 <= perc <= 20:
        return "low"
    elif 20 < perc <= 60:
        return "medium"
    elif 60 < perc <= 100:
        return "high"

def baticon(perc):
          #   0   10   20   30   40   50   60   70   80   90   100
    icons = ['󱃍', '󰁺', '󰁻', '󰁼', '󰁽', '󰁾', '󰁿', '󰂀', '󰂁', '󰂂', '󰁹']
    return icons[perc // 10]

if __name__ == "__main__":
    perc, timerem = battperc()
    d = {
        "icon":       baticon(perc), 
        "percentage": perc,
        "timerem":    timerem,
        "class":      batclass(perc)
    }
    print(json.dumps(d))

File: eww/go/bat.go
package main

import (
	"encoding/json"
	"fmt"
	"os/exec"
	"strconv"
	"strings"
)

func battperc() (int, string, error) {
	cmd := exec.Command("acpi", "-b")
	output, err := cmd.Output()
	if err != nil {
		return 0, "", err
	}

	data := strings.Split(string(output), ",")[1:3]
	perc, err := strconv.Atoi(strings.ReplaceAll(strings.TrimSpace(data[0]), "%", ""))
	if err != nil {
		return 0, "", err
	}
	timerem := strings.ReplaceAll(strings.ReplaceAll(strings.TrimSpace(data[1]), "remaining", ""), "until charged", "")
	timerem = strings.ReplaceAll(strings.TrimSpace(timerem), " ", "")
	return perc, timerem[:len(timerem)-3], nil
}

func batclass(perc int) string {
	if perc >= 0 && perc <= 20 {
		return "low"
	} else if perc > 20 && perc <= 60 {
		return "medium"
	} else if perc > 60 && perc <= 100 {
		return "high"
	}
	return "unknown"
}

func baticon(perc int) string {
	icons := []string{"󱃍", "󰁺", "󰁻", "󰁼", "󰁽", "󰁾", "󰁿", "󰂀", "󰂁", "󰂂", "󰁹"}
	return icons[perc/10]
}

type BatteryStatus struct {
	Icon       string `json:"icon"`
	Percentage int    `json:"percentage"`
	Timerem    string `json:"timerem"`
	Class      string `json:"class"`
}

func main() {
	perc, timerem, err := battperc()
	if err != nil {
		fmt.Println(err)
		return
	}
	d := BatteryStatus{
		Icon:       baticon(perc),
		Percentage: perc,
		Timerem:    timerem,
		Class:      batclass(perc),
	}
	jsonData, err := json.Marshal(d)
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Println(string(jsonData))
}
